default-recommendations/for-loop-shortcuts-test.rkt

```diff
@@ -909,3 +909,73 @@ test: "(unless ...) in a for* loop refactored to #:when clause"
        #:unless (number? x))
   (displayln x))
 ------------------------------------------------------------
+
+
+test: "for/vector with in-range 0 n gets #:length n"
+------------------------------------------------------------
+(define n 5)
+(for/vector ([i (in-range 0 n)]) i)
+------------------------------------------------------------
+------------------------------------------------------------
+(define n 5)
+(for/vector #:length n
+            ([i (in-range 0 n)])
+  i)
+------------------------------------------------------------
+
+
+test: "for/vector with in-range n gets #:length n"
+------------------------------------------------------------
+(define n 5)
+(for/vector ([i (in-range n)]) i)
+------------------------------------------------------------
+------------------------------------------------------------
+(define n 5)
+(for/vector #:length n
+            ([i (in-range 0 n)])
```

Comment:
The input was written as `(in-range n)`, so the output should be written the same way. The rule should preserve the entire `(in-range ...)` expression exactly as it was written in the input.

```suggestion
            ([i (in-range n)])
```